### PR TITLE
Add `.mypy.ini`, `pylintrc` and satisfy `pylint`, `mypy` checks

### PR DESCRIPTION
**Pull request by Yo1k**

### CODE REVIEW

File: yo1k/tic_tac_toe/core/util.py [hunk 1]
```diff
@@ -1,6 +1,12 @@
-def eq(cls):
+from __future__ import annotations
+from typing import TypeVar, Type
+
+T = TypeVar("T")  # pylint: disable=C0103
```
Comment (Yo1k) [inline yo1k/tic_tac_toe/core/util.py:4]: `Pylint 2.12` (or earlier versions) [does not work properly](https://github.com/PyCQA/pylint/issues/3401) with `TypeVar` in the  (it is fixed in the 2.13).

File: yo1k/tic_tac_toe/core/util.py [hunk 2]
```diff
@@ -1,6 +1,12 @@
...
+
+
+def eq(cls: Type[T]) -> Type[T]:
     """Class decorator providing generic comparison functionality."""
-    def __eq__(self, other) -> bool:
+    def __eq__(self: T, other: object) -> bool:
         return isinstance(other, self.__class__) and self.__dict__ == other.__dict__
-    cls.__eq__ = __eq__
+    cls.__eq__ = __eq__  # type: ignore
```
Comment (Yo1k) [inline yo1k/tic_tac_toe/core/util.py:11]: `Mypy` do not allow assigning a function to a method (see [here](https://github.com/python/mypy/issues/2427)).

File: yo1k/tic_tac_toe/test/core/test_game.py [hunk 3]
```diff
@@ -112,10 +112,10 @@ def test_win_condition(self):
                  Cell(0, 2), False)]
         for cells, cell, expect in args_and_expect_list:
```
Comment (Yo1k) [inline yo1k/tic_tac_toe/test/core/test_game.py:113]: `Mypy` does not inference the right type for `cells` and the syntax [does not allow one to annotate the types of variables when tuple unpacking is used](https://www.python.org/dev/peps/pep-0526/#:~:text=Note%20that%2C%20although%20the%20syntax%20does%20allow%20tuple%20packing%2C%20it%20does%20not%20allow%20one%20to%20annotate%20the%20types%20of%20variables%20when%20tuple%20unpacking%20is%20used%3A).

File: yo1k/tic_tac_toe/core/util.py (hunk 1 above)
Comment (stIncMale) [reply yo1k/tic_tac_toe/core/util.py:4]: Could you please create a GitHub issue to update to Pylint 2.13 when it comes out, and to remove the suppression here?

File: .pylintrc [hunk 4]
```diff
@@ -0,0 +1,579 @@
+[MASTER]
```
Comment (stIncMale) [inline .pylintrc:1]: Was this Pylint config auto-generated, or grabbed somewhere from Pylint?

File: yo1k/tic_tac_toe/test/core/test_game.py [hunk 5]
```diff
@@ -8,14 +11,11 @@
...
-from typing import Optional
-import unittest
 
 
 class ListActionQueue(ActionQueue):
-    def __init__(self, actions: Sequence[Optional[Action]]):
-        self.actions = actions
+    def __init__(self, actions: MutableSequence[Optional[Action]]):
+        self.actions: MutableSequence[Optional[Action]] = actions
         self.i = 0
```
Comment (stIncMale) [inline yo1k/tic_tac_toe/test/core/test_game.py]: The attribute type annotation is missing for `self.i`.

File: yo1k/tic_tac_toe/test/core/test_game.py [hunk 6]
```diff
@@ -112,10 +112,10 @@ def test_win_condition(self):
                  Cell(0, 2), False)]
         for cells, cell, expect in args_and_expect_list:
             with self.subTest(expect=expect, cell=cell, cells=cells):
-                actual = Logic._win_condition(Board(cells), cell)
+                actual = Logic.win_condition(Board(cells), cell)  # type: ignore
```
Comment (stIncMale) [inline yo1k/tic_tac_toe/test/core/test_game.py:115]: What is the reasoning behind making this static method public?

File: yo1k/tic_tac_toe/core/game.py [hunk 7]
```diff
@@ -179,13 +182,13 @@ def next(self) -> Optional[Action]:
 class Logic:
     """Game logic."""
 
-    def __init__(self, action_queues: Sequence[ActionQueue]):
+    def __init__(self, action_queues: MutableSequence[ActionQueue]):
```
Comment (stIncMale) [inline yo1k/tic_tac_toe/core/game.py]: Why does this parameter need to be `MutableSequence`? `Logic` is not supposed to either add or remove any elements from it.

**Review by stIncMale — CHANGES_REQUESTED** [review]
Good job 👍  Just a few little things.

File: yo1k/tic_tac_toe/core/game.py [hunk 8]
```diff
@@ -256,23 +259,23 @@ def __last_step(step: int) -> bool:
         return step == Board.size() ** 2 - 1
 
     @staticmethod
-    def __surrender(state: State):
+    def __surrender(state: State) -> None:
         assert state.phase is Phase.INROUND
         idx_other_player = (state.turn() + 1) % 2
```
Comment (stIncMale) [inline yo1k/tic_tac_toe/core/game.py]: Use `len(state.players)` instead of 2.

File: yo1k/tic_tac_toe/core/game.py [hunk 9]
```diff
@@ -256,23 +259,23 @@ def __last_step(step: int) -> bool:
         return step == Board.size() ** 2 - 1
 
     @staticmethod
-    def __surrender(state: State):
+    def __surrender(state: State) -> None:
         assert state.phase is Phase.INROUND
```
Comment (stIncMale) [inline yo1k/tic_tac_toe/core/game.py:260]: Add another assertion:
```python
# for more players this method would have been implemented quite differently
assert State.player_count() is 2
```

File: yo1k/tic_tac_toe/core/game.py [hunk 10]
```diff
@@ -35,45 +35,47 @@ class Cell:
...
 
-    def set(self, cell: Cell, mark: Mark):
+    def set(self, cell: Cell, mark: Mark) -> None:
         self.cells[cell.x][cell.y] = mark
 
     def get(self, cell: Cell) -> Optional[Mark]:
         return self.cells[cell.x][cell.y]
 
-    def clean(self):
+    def clean(self) -> None:
```
Comment (stIncMale) [inline yo1k/tic_tac_toe/core/game.py]: Let's rename `clean` -> `clear`, to be consistent with [`MutableSequence.clear`](https://docs.python.org/3/library/stdtypes.html#mutable-sequence-types)

File: yo1k/tic_tac_toe/core/game.py [hunk 11]
```diff
@@ -216,22 +219,22 @@ def __advance_inround(self, state: State):
...
 
     @staticmethod
-    def __occupy(state: State, cell: Cell):
+    def __occupy(state: State, cell: Cell) -> None:
         assert state.phase is Phase.INROUND
         mark = state.players[state.turn()].mark
         if state.board.get(cell) is None:
             state.board.set(cell, mark)
         else:
             assert False
```
Comment (stIncMale) [inline yo1k/tic_tac_toe/core/game.py]: Replace
```python
if state.board.get(cell) is None:
    state.board.set(cell, mark)
else:
    assert False
```
with
```python
state.board.set(cell, mark)
```
and add the following assertion to `Board.set`:
```python
assert self.cells[cell.x][cell.y] is None
```

File: yo1k/tic_tac_toe/core/util.py (hunk 1 above)
Comment (Yo1k) [reply yo1k/tic_tac_toe/core/util.py:4]: Done.

File: .pylintrc (hunk 4 above)
Comment (Yo1k) [reply .pylintrc:1]: Initially this config was auto-generated and then I added some changes like:
1) add notes to the `disable` in the `[MESSAGES CONTROL]` section
2) add notes to the  `[MISCELLANEOUS]` section
3) add names to `good-names` in the `[BASIC]` section

File: yo1k/tic_tac_toe/test/core/test_game.py (hunk 5 above)
Comment (Yo1k) [reply yo1k/tic_tac_toe/test/core/test_game.py]: Done.

File: yo1k/tic_tac_toe/core/game.py [hunk 12]
```diff
@@ -287,24 +290,24 @@ def __ready(state: State, player_idx: int):
             state.phase = Phase.INROUND
 
     @staticmethod
-    def __set_outround(state: State):
+    def __set_outround(state: State) -> None:
```
Comment (stIncMale) [inline yo1k/tic_tac_toe/core/game.py]: Let's rename this method to `__end_round`?

File: yo1k/tic_tac_toe/test/core/test_game.py (hunk 6 above)
Comment (Yo1k) [reply yo1k/tic_tac_toe/test/core/test_game.py:115]: As I mentioned before, we can not use protected static method from one module to another, even if it is used inside unittests (both our chekers rise error). And we use this method to test win condition algorithm .

File: yo1k/tic_tac_toe/core/game.py (hunk 8 above)
Comment (Yo1k) [reply yo1k/tic_tac_toe/core/game.py]: Done.

File: yo1k/tic_tac_toe/core/game.py (hunk 9 above)
Comment (Yo1k) [reply yo1k/tic_tac_toe/core/game.py:260]: Done.

File: yo1k/tic_tac_toe/core/game.py (hunk 10 above)
Comment (Yo1k) [reply yo1k/tic_tac_toe/core/game.py]: Done.

File: yo1k/tic_tac_toe/core/game.py (hunk 11 above)
Comment (Yo1k) [reply yo1k/tic_tac_toe/core/game.py]: Done.

File: yo1k/tic_tac_toe/core/game.py (hunk 12 above)
Comment (Yo1k) [reply yo1k/tic_tac_toe/core/game.py]: Done.

File: yo1k/tic_tac_toe/core/game.py (hunk 7 above)
Comment (Yo1k) [reply yo1k/tic_tac_toe/core/game.py]: Done.

File: .pylintrc (hunk 4 above)
Comment (stIncMale) [reply .pylintrc:1]: Let's discuss `disable` in `[MESSAGES CONTROL]` and `notes` in `[MISCELLANEOUS]`.

File: yo1k/tic_tac_toe/core/game.py (hunk 9 above)
Comment (stIncMale) [reply yo1k/tic_tac_toe/core/game.py:260]: I left a comment in the example for a reason: I think we need to clarify this new assertion.

**Review by stIncMale — CHANGES_REQUESTED** [review]

File: yo1k/tic_tac_toe/core/game.py (hunk 9 above)
Comment (Yo1k) [reply yo1k/tic_tac_toe/core/game.py:260]: Done.

File: .pylintrc (hunk 4 above)
Comment (Yo1k) [reply .pylintrc:1]: Done.

**Review by stIncMale — APPROVED** [review]

**Pull request merged**